lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -149,7 +149,7 @@ class ReportRenderer {
     const warnings = this._dom.find('ul', container);
     for (const warningString of report.runWarnings) {
       const warning = warnings.appendChild(this._dom.createElement('li'));
-      warning.textContent = warningString;
+      warning.appendChild(this._dom.convertMarkdownLinkSnippets(warningString));
```

Comment:
The only warning I could find that allowed arbitrary-ish user input was in the redirect warning that includes the redirect URL. That message already won't trigger the markdown link snippets and even if it did we'd only be including a link to the same URL we already link to at the very top of the report.

If folks are really concerned about this as some sort of attack vector, we could limit the allowed URLs to web.dev/developers.google.com/github.com or something?

Reply:
> If folks are really concerned about this as some sort of attack vector, we could limit the allowed URLs to web.dev/developers.google.com/github.com or something?

While writing this I realized we want plugins to be able to link to their docs, so I don't think we can do this idea.

Looking over
- [DOM_based_XSS_Prevention_Cheat_Sheet](https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html)
- [Cross_Site_Scripting_Prevention_Cheat_Sheet](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md#rule-5---url-encode-before-inserting-untrusted-data-into-html-url-parameter-values)

I believe we are ok. `href` is the only thing we're setting, we prevent `javascript:` urls (really anything that isn't `/^https?:/`), and we actually pass the url through the `URL` constructor.

Are there any differences between setting `href` as a property and through `setAttribute()`? OWASP does seem to prefer the second way, but I don't know if it's material for the way we're creating the report vs the server-side templatey stuff they have in all of their examples.

Reply:
(maybe there's an additional test or two we could add to make sure we don't ever quietly regress on more of this stuff, but I don't have any immediate ideas)

Reply:
> I realized we want plugins to be able to link to their docs

I probably wasn't very clear but my suggestion was to have this filter param be optional in that method and only use it for the toplevel warnings. Individual audits in plugins would still be able to link to their docs (and stack packs, etc). 

I'm glad we don't have to deal with this though, even better :D

I added a few more cases to the existing markdown link tests.